src/mlir/generator.cc

```diff
@@ -356,24 +404,130 @@ namespace mlir::verona
     }
 
     // Else, it should be an operation that we can lower natively
-    // TODO: Separate between unary, binary, ternary, etc.
-    // FIXME: Make this able to discern different types of operations.
-    if (name == "+")
+    if (isUnary(ast))
+    {
+      return parsingError(
+        "Unary Operation '" + name + "' not implemented yet", getLocation(ast));
+    }
+    else if (isBinary(ast))
     {
+      // Get bpth arguments
       auto arg0 = parseNode(getOperand(ast, 0).lock());
       if (auto err = arg0.takeError())
         return std::move(err);
       auto arg1 = parseNode(getOperand(ast, 1).lock());
       if (auto err = arg1.takeError())
         return std::move(err);
-      auto dialect = Identifier::get("type", &context);
-      auto type = mlir::OpaqueType::get(dialect, "ret", &context);
-      return genOperation(getLocation(ast), "verona.add", {*arg0, *arg1}, type);
+      if (name == "+")
```

Comment:
Would this nor be simpler as a `StringSwitch` returning a pair, if it's just mapping from `name` to the op name and type?